### PR TITLE
viking: Add "default_version" to `config.toml`

### PR DESCRIPTION
**Pull request by MonsterDruide1**
As addition to the last PR, which added the `--version` parameter, this PR includes a new entry into the `config.toml`, which allows to supply a `default_version`.

Games like SMO, which should *only* have version-specific files, it might get tedious to re-enter the version on each command. To simplify the work there, a default version can be added into the config file, which will be used when no `--version` parameter is used in the call.

If neither a parameter nor a `default_version` is given, it will just default to the default build and data directory without version numbers.

---
This change is [Reviewable](https://reviewable.io/reviews/open-ead/nx-decomp-tools/6)

### CODE REVIEW

File: viking/src/tools/check.rs [hunk 1]
```diff
@@ -174,11 +174,13 @@ fn get_function_to_check_from_args(args: &[String]) -> Result<String> {
...
 }
 
-fn get_version_from_args(args: &[String]) -> Result<Option<&str>> {
+fn get_version_from_args_or_config(args: &[String]) -> Result<Option<&str>> {
     let mut iter = args.iter().filter_map(|s| s.strip_prefix("--version="));
     match (iter.next(), iter.next()) {
         (Some(_), Some(_)) => bail!("expected only one version number ('--version=XXX')"),
-        (None, None) => Ok(None),
+        (None, None) => Ok(
+            repo::CONFIG.get("default_version").map(|s| s.as_str()).unwrap_or_else(|| None)
```
Comment (leoetlino) [inline viking/src/tools/check.rs:182]: `.unwrap_or_else(|| None)` -> `ok()`

**Review by leoetlino — CHANGES_REQUESTED** [review]

File: viking/src/tools/check.rs (hunk 1 above)
Comment (MonsterDruide1) [reply viking/src/tools/check.rs:182]: That would map `None` to `Err(...)`, which is not correct. Instead, it should unwrap to `None` in that case.

The problem is that `.get()` and then `as_str()` seems to return an `Option<Option<&str>>`, so one instance has to be unwrapped there.

Comment (leoetlino) [reply viking/src/tools/check.rs:182]: Oh, so `repo::CONFIG.get("default_version").map(|s| s.as_str())` is an `Option<Option<...>>`. Thought it was a Result.

`.unwrap_or_else(|| None)` still looks rather silly though -- `.unwrap_or(None)` would be preferable.

Comment (MonsterDruide1) [reply viking/src/tools/check.rs]: Done.

**Review by leoetlino — APPROVED** [review]

**Pull request merged**